components/elm/src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -492,7 +524,31 @@ subroutine Infiltration(bounds, num_hydrologyc, filter_hydrologyc, num_urbanc, f
              !7. remove drainage from h2osfc and add to qflx_infl
              h2osfc(c) = h2osfc(c) - qflx_h2osfc_drain(c) * dtime
              qflx_infl(c) = qflx_infl(c) + qflx_h2osfc_drain(c)
-             qflx_gross_infl_soil(c) = qflx_gross_infl_soil(c) + qflx_h2osfc_drain(c)
+             
+             !8. add drainage from river inundation to qflx_infl (land river two way coupling)
+             if (use_lnd_rof_two_way) then
+
+               h2osoi_left_vol1 = max(max(0._r8,(pondmx+watsat(c,1)*dz(c,1)*1.e3_r8-h2osoi_ice(c,1)-watmin)) - &
```

Comment:
can you please simplify the code so there aren't so many nested `max` as it is hard to follow the code?

Reply:
I added a if statement to remove one max().